service/awsclusterconfig/framework.go

```diff
@@ -0,0 +1,125 @@
+package awsclusterconfig
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/informer"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	"github.com/giantswarm/cluster-operator/service/awsclusterconfig/v1"
+)
+
+// FrameworkConfig contains necessary dependencies and settings for
+// AWSClusterConfig CRD framework implementation.
+type FrameworkConfig struct {
+	BaseClusterConfig *cluster.Config
+	G8sClient         versioned.Interface
+	K8sClient         kubernetes.Interface
+	K8sExtClient      apiextensionsclient.Interface
+	Logger            micrologger.Logger
+
+	ProjectName string
+}
+
+// NewFramework returns a configured AWSClusterConfig framework implementation.
+func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.G8sClient must not be empty")
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.K8sExtClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sExtClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.ProjectName must not be empty")
+	}
+
+	var err error
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.Config{
+			K8sExtClient: config.K8sExtClient,
+			Logger:       config.Logger,
+		}
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.Config{
+			Watcher: config.G8sClient.CoreV1alpha1().AWSClusterConfigs(""),
+		}
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var v1ResourceSet *framework.ResourceSet
+	{
+		c := v1.ResourceSetConfig{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			HandledVersionBundles: []string{
+				"0.1.0",
+			},
```

Comment:
I would like to align this with the other operators. We do not need to configure this list. We should just implement the handle func based on the local version bundle in the resource set.

Reply:
Will fix.

Reply:
Just to verify: You mean here that this should be removed and then rely on resource set handle func? I'm asking this because in both, `kvm-operator` and  `aws-operator` there are spots where this is defined:
- https://github.com/giantswarm/kvm-operator/blob/master/service/kvmconfig/framework.go#L117
- https://github.com/giantswarm/aws-operator/blob/master/service/awsconfig/cluster_framework.go#L228
- https://github.com/giantswarm/aws-operator/blob/master/service/awsconfig/cluster_framework.go#L254
- https://github.com/giantswarm/aws-operator/blob/master/service/awsconfig/cluster_framework.go#L278
- ...

I presume that these are just legacy from earlier design and should be boyscouted away at some point?

Reply:
Yes because you check early versions. Check the latest https://github.com/giantswarm/kvm-operator/blob/f17b1a7349a026fd7de1d398fe1d63a9b0029cb1/service/kvmconfig/framework.go#L257 No `HandleFunc` there.

Reply:
What Pawel said. We did not backport all earlier versions for this because they will be gone eventually anyway. The latest versions define our standards, if implemented.